libvast/vast/index_config.hpp

```diff
@@ -0,0 +1,53 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/type.hpp"
+
+#include <string>
+#include <vector>
+
+namespace vast {
+
+/// The configuration for `vast.index`.
+struct index_config {
+  static constexpr bool use_deep_to_string_formatter = true;
+
+  struct rule {
+    std::vector<std::string> targets;
+    double fp_rate = 0.01;
+
+    template <class Inspector>
+    friend auto inspect(Inspector& f, rule& x) {
+      return f(x.targets, x.fp_rate);
+    }
+
+    static inline const record_type& layout() noexcept {
+      static auto result = record_type{
+        {"targets", list_type{string_type{}}},
+        {"fp-rate", real_type{}},
+      };
+      return result;
+    }
+  };
+
+  std::vector<rule> rules;
```

Comment:
```suggestion
  std::vector<rule> rules = {};
```

Same rationale as above.